docs/synthesis.md

```diff
@@ -15,10 +20,45 @@ synthesis:
 | -------- | -------- | ------------ | ---------------- |
 | name    | String | Yes | The attribute to use for the entity name. |
 | identifier| String| Yes | Telemetry attribute to use as the entity identifier.|
+| compositeIdentifier| String| No | Set of attributes that will identify the telemetry. When this one is used identifier is not required.|
 | encodeIdentifierInGUID | Boolean | No | If true, the identifier value will be encoded to respect the [GUID limits][guid_spec]. Defaults to `false`. |
 | conditions | List | No | The list of conditions to apply in the data point to match the rule. Defaults to an empty list. |
 | tags     | List   | No | The list of attributes to copy as entity tags if the rule matches. Defaults to an empty list. |
 
+### Identifier
+
+As previously mentioned the identifier should be a unique value for that entityType in a specific user account.
+
+For example if a service identifier is the `serviceName` attribute, multiple applications reporting the same value under the same account will be treated as only one entity.
+
+The identifier is the most important piece of information of an entity since changing it means creating a new entity and hence not having the previous produced telemetry linked to the new entity.
+
+#### Composite identifier
+
+The general advice is to use only one attribute as the identifier but there are some situations where only one attribute is not enough to uniquely identify the entity.
+
+In these cases `compositeIdentifier` can be used to define multiple attributes as the identifier, and a separator between each attribute.
+Also we recommend to always set `encodeIdentifierInGUID: true` when using `compositeIdentifier`
+
+```yaml
+synthesis:
+  rules:
+  - compositeIdentifier:
+      separator: "/"
+      attributes:
+        - k8s.namespace
+        - k8s.deployment
+```
+
+There are a few drawbacks when using composite identifiers:
```

Comment:
Nit: It will be simple to read these drawbacks if you add them as a list

Reply:
add the list